backend/satellite_tools/reposync.py

```diff
@@ -1829,7 +1834,9 @@ def _url_with_repo_credentials(self, url_in):
             url.password = base64.decodestring(credentials['password'].encode()).decode()
             # remove query parameter from url
             url.query = ""
-        return url.getURL()
+            if 'extra_auth' in credentials and credentials['extra_auth']:
+                headers = json.loads(credentials['extra_auth'].tobytes())
```

Comment:
Only 1 question left: must `extra_auth` really be a byte array/blob in DB or could it be also just "TEXT".
Might be easier to handle. But now everything is written and tested we might better stay with it.

Reply:
It could be just text, or it can be even Json or Jsonb. I added it as byte array so it can be more flexible in the future and add anything that could be needed.
If you think is best, I can added change to text or json and adapt the code.

Reply:
keep it as bytes. This is now tested and all places are aligned.